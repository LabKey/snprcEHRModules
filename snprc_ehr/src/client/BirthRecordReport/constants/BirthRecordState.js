const birthRecordState = () => {
    return ({
        isDirty: false,
        summaryData: [],
        animalList: [],
        isLoading: true,
        selectedAnimal: undefined,
        errorMessage: undefined
    })
}

export default birthRecordState
