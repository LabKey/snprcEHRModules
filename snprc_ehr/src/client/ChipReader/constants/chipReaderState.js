const ChipReaderState = () => {
    return ({
        port: undefined,
        serialOptions: [],
        connection: undefined,
        chipData: { chipId: undefined, animalId: undefined, temperature: undefined, time: undefined },
        summaryData: [],
        isLoading: true,
        hasError: false,
        showCancelModal: false,
        errorMessage: undefined
    })
}

export default ChipReaderState
