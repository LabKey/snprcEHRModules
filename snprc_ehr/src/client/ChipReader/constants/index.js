export default {
    debug: false,
    notSupportedMessage: `Sorry, Serial.API is not supported on this device, make sure you're 
    running Chrome 78 or later and have enabled the #enable-experimental-web-platform-features flag in
    chrome://flags`,
    timeOutErrorMessage: 'Chip reader is not responding. Ensure you are connected to the correct port, and the reader is turned on.',
    validSerialOptions: [ 'baudrate', 'databits', 'stopbits', 'parity', 'buffersize', 'rtscts', 'xon', 'xoff', 'xany' ],
    defaultSerialOptions: {
        baudrate: 19200,
        databits: 8,
        stopbits: 1,
        buffersize: 255,
        parity: 'none',
        rctscts: false,
        xon: true,
        xoff: true,
        xany: false
    }
}

// https://wicg.github.io/serial/#dom-serial
// parity is an enum:   "none",  "even",  "odd",  "mark",  "space"
// baudrate member: One of 115200, 57600, 38400, 19200, 9600, 4800, 2400, 1800, 1200, 600, 300, 200, 150, 134, 110, 75, or 50.
// databits member: One of 8, 7, 6, or 5.
// stopbits member: One of 1 or 2.
// rctscts, xon, xoff, xany: boolean