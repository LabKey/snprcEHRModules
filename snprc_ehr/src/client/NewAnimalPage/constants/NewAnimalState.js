import moment from 'moment';

const NewAnimalState = () => {

    return  ({
        currentPanel: 1,
        selectedOption: undefined,
        newAnimalData: {
            selectedOption: undefined,
            id: undefined,
            birthDate: { date: moment() },
            bdStatus: undefined,
            acquisitionType: undefined,
            acqDate: { date: moment() },
            gender: undefined,
            sire: undefined,
            dam: undefined,
            species: undefined,
            colony: undefined,
            animalAccount: undefined,
            ownerInstitution: undefined,
            responsibleInstitution: undefined,
            room: undefined,
            cage: { value: undefined },
            diet: undefined,
            pedigree: undefined,
            iacuc: undefined
        },
        summaryData: [],
        speciesList: [],
        acquisitionTypeList: [],
        potentialDamList: [],
        potentialSireList: [],
        locationList: [],
        accountList: [],
        colonyList: [],
        dietList: [],
        institutionList: [],
        iacucList: [],
        pedigreeList: [],
        bdStatusList: [],
        isLoading: true,
        hasError: false,
        preventNext: true,
        saveOk: false,
        showSaveModal: false,
        showCancelModal: false,
        showSpeciesChangeModal: false,
        errorMessage: undefined
    });
}

export default NewAnimalState;
