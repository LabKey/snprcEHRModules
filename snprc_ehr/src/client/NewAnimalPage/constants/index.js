export default {
    debug: false,
    numPanels: 5,
    defaultInstitution: { id: 0, value: 1, label: 'TxBiomed - Texas Biomedical Research Institute' },
    offSiteAcqCodes: ['22', '23', '25', '97'],
    hamsterWarnings:
    `Hamsters can be acquired in bulk; however, when added in bulk you must ensure all animals:
        1) are the same sex
        2) have the same birth and acquisition dates
        3) are in the same location
        4) are on the same account, colony, and IACUC `
}
