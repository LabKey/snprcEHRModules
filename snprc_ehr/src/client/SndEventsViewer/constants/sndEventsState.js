import defaults from './index'

const SndEventsState = () => {

    return ({
        isDirty: false,
        summaryData: [],
        isLoading: true,
        hasError: false,
        showCancelModal: false,
        errorMessage: undefined
    })
}

export default SndEventsState
