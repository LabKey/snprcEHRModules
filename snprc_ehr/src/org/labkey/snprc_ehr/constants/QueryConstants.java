package org.labkey.snprc_ehr.constants;

import org.apache.commons.text.CaseUtils;

/**
 * String constants used for database queries. Contains table names, schema names, column names, SQL aueries,
 * and other database query information
 */
public class QueryConstants
{
    public static final String AGE_AT_TIME_COLUMN = "Age At Time";
    public static final String AGE_AT_TIME_YEARS_COLUMN = "Age At Time Years";
    public static final String AGE_AT_TIME_YEARS_ROUNDED_COLUMN = "Age At Time Years Rounded";
    public static final String AGE_AT_TIME_DAYS_COLUMN = "Age At Time Days";
    public static final String AGE_AT_TIME_MONTHS_COLUMN = "Age At Time Months";
    public static final String AGE_CLASS_AT_TIME_COLUMN = "Age Class At Time";
    public static final String ASSIGNMENT_AT_TIME_COLUMN = "Assignment At Time";
    public static final String STUDY_SCHEMA = "study";
    public static final String CODE_COLUMN = "code";
    public static final String SPECIES_TABLE = "species";
    public static final String PROJECT_TABLE = "project";
    public static final String ANIMAL_TABLE = "animal";
    public static final String ASSIGNMENT_TABLE = "assignment";
    public static final String DATE_COLUMN = "date";
    public static final String DEMOGRAPHICS_TABLE = "demographics";
    public static final String ID_COLUMN = "Id";
    public static final String ROW_ID_COLUMN = "rowid";
    public static final String PROTOCOLS_AT_TIME_CALCULATED = "Protocols At Time";
    public static final String ACCOUNTS_AT_TIME_CALCULATED = "Accounts At Time";
    public static final String PRIMARY_KEY_VARIABLE = "primaryKeyColumnName";
    public static final String SCHEMA_VARIABLE = "schemaName";
    public static final String EHR_PATH_VARIABLE = "ehrPath";
    public static final String QUERY_VARIABLE = "queryName";
    public static final String ID_COLUMN_VARIABLE = "idColumnName";
    public static final String TARGET_CONTAINER_VARIABLE ="targetContainerName";
    public static final String DATE_COLUMN_VARIABLE = "dateColumnName";
    public static final String EHR_LOOKUPS_SCHEMA = "ehr_lookups";
    public static final String AGE_CLASS_LOOKUP_URL = "/detailsQueryRow.view?schemaName=ehr_lookups&query.queryName=ageclass&rowid=";


    public static final String AGE_CLASS_TABLE = "ageClass";
    public static final String ASSIGNMENT_AT_TIME_SQL = "SELECT\n" +
            "sd.${" + PRIMARY_KEY_VARIABLE + "},\n" +
            "group_concat(DISTINCT h.protocol.displayName, chr(10)) as " + CaseUtils.toCamelCase(PROTOCOLS_AT_TIME_CALCULATED, true) + ",\n" +
            "group_concat(DISTINCT ao.account.account, chr(10)) as " + CaseUtils.toCamelCase(ACCOUNTS_AT_TIME_CALCULATED, true) + ",\n" +
            "FROM \"${" + SCHEMA_VARIABLE + "}\".\"${" + QUERY_VARIABLE + "}\" sd\n" +
            "JOIN \"${" + TARGET_CONTAINER_VARIABLE + "}\".study.assignment h\n" +
            "  ON (sd.id = h.id AND h.dateOnly <= CAST(sd.${" + DATE_COLUMN_VARIABLE + "} AS DATE) AND (CAST(sd.${" + DATE_COLUMN_VARIABLE +"} AS DATE) <= h.enddateCoalesced) AND h.qcstate.publicdata = true)\n" +
            "JOIN \"${" + TARGET_CONTAINER_VARIABLE + "}\".study.animalAccounts ao\n" +
            "  ON (sd.id = ao.id AND ao.dateOnly <= CAST(sd.${" + DATE_COLUMN_VARIABLE + "} AS DATE) AND (CAST(sd.${" + DATE_COLUMN_VARIABLE +"} AS DATE) <= ao.enddateCoalesced) AND ao.qcstate.publicdata = true)\n" +
            "group by sd.${"+ PRIMARY_KEY_VARIABLE + "}";

    public static final String AGE_AT_TIME_SQL = "SELECT\n" +
            "c.${" + PRIMARY_KEY_VARIABLE + "},\n" +
            "\n" +
            "CAST(\n" +
            "CASE\n" +
            "WHEN d.birth is null or c.${" + DATE_COLUMN_VARIABLE + "} is null\n" +
            "  THEN null\n" +
            "WHEN (d.lastDayAtCenter IS NOT NULL AND d.lastDayAtCenter < c.${" + DATE_COLUMN_VARIABLE + "}) THEN\n" +
            " ROUND(CONVERT(age_in_months(d.birth, d.lastDayAtCenter), DOUBLE) / 12, 1)\n" +
            "ELSE\n" +
            "  ROUND(CONVERT(age_in_months(d.birth, CAST(c.${" + DATE_COLUMN_VARIABLE + "} as DATE)), DOUBLE) / 12, 1)\n" +
            "END AS float) as " + CaseUtils.toCamelCase(AGE_AT_TIME_COLUMN, true) + ",\n" +
            "\n" +

            "CAST(\n" +
            "CASE\n" +
            "WHEN d.birth is null or c.${" + DATE_COLUMN_VARIABLE + "} is null\n" +
            "  THEN null\n" +
            "WHEN (d.lastDayAtCenter IS NOT NULL AND d.lastDayAtCenter < c.${" + DATE_COLUMN_VARIABLE + "}) THEN\n" +
            " ROUND(CONVERT(timestampdiff('SQL_TSI_DAY', d.birth, d.lastDayAtCenter), DOUBLE) / 365.25, 2)\n" +
            "ELSE\n" +
            "  ROUND(CONVERT(timestampdiff('SQL_TSI_DAY', d.birth, CAST(c.${" + DATE_COLUMN_VARIABLE + "} as DATE)), DOUBLE) / 365.25, 2)\n" +
            "END AS float) as " + CaseUtils.toCamelCase(AGE_AT_TIME_YEARS_COLUMN, true) + ",\n" +
            "\n" +
            "CAST(\n" +
            "CASE\n" +
            "WHEN d.birth is null or c.${" + DATE_COLUMN_VARIABLE + "} is null\n" +
            "  THEN null\n" +
            "WHEN (d.lastDayAtCenter IS NOT NULL AND d.lastDayAtCenter < c.${" + DATE_COLUMN_VARIABLE + "}) THEN\n" +
            " floor(age(d.birth, d.lastDayAtCenter))\n" +
            "ELSE\n" +
            "  floor(age(d.birth, CAST(c.${" + DATE_COLUMN_VARIABLE + "} as DATE)))\n" +
            "END AS float) as " + CaseUtils.toCamelCase(AGE_AT_TIME_YEARS_ROUNDED_COLUMN, true) + ",\n" +
            "\n" +
            //Added 'Age at time Days' by kollil on 02/15/2019
            "CAST(\n" +
            "CASE\n" +
            "WHEN d.birth is null or c.${" + DATE_COLUMN_VARIABLE + "} is null\n" +
            "  THEN null\n" +
            "WHEN (d.lastDayAtCenter IS NOT NULL AND d.lastDayAtCenter < c.${" + DATE_COLUMN_VARIABLE + "}) THEN\n" +
            "  CONVERT(TIMESTAMPDIFF('SQL_TSI_DAY',d.birth, d.lastDayAtCenter), INTEGER)\n" +
            "ELSE\n" +
            "  CONVERT(TIMESTAMPDIFF('SQL_TSI_DAY',d.birth, CAST(c.${" + DATE_COLUMN_VARIABLE + "} AS DATE)), INTEGER)\n" +
            "END AS float) as " + CaseUtils.toCamelCase(AGE_AT_TIME_DAYS_COLUMN, true) + ",\n" +
            "\n" +
            //
            "CAST(\n" +
            "CASE\n" +
            "WHEN d.birth is null or c.${" + DATE_COLUMN_VARIABLE + "} is null\n" +
            "  THEN null\n" +
            "WHEN (d.lastDayAtCenter IS NOT NULL AND d.lastDayAtCenter < c.${" + DATE_COLUMN_VARIABLE + "}) THEN\n" +
            "  CONVERT(age_in_months(d.birth, d.lastDayAtCenter), INTEGER)\n" +
            "ELSE\n" +
            "  CONVERT(age_in_months(d.birth, CAST(c.${" + DATE_COLUMN_VARIABLE + "} AS DATE)), INTEGER)\n" +
            "END AS float) as " + CaseUtils.toCamelCase(AGE_AT_TIME_MONTHS_COLUMN, true) + ",\n" +
            //NOTE: written as subselect so we ensure a single row returned in case data in ehr_lookups.ageclass has rows that allow dupes
            "(SELECT ac.label FROM ehr_lookups.ageclass ac\n" +
            "  WHERE " +
            "  (CONVERT(age_in_months(d.birth, COALESCE(d.lastDayAtCenter, now())), DOUBLE) / 12) >= ac.\"min\" AND\n" +
            "  ((CONVERT(age_in_months(d.birth, COALESCE(d.lastDayAtCenter, now())), DOUBLE) / 12) < ac.\"max\" OR ac.\"max\" is null) AND\n" +
            "  d.species.arc_species_code = ac.species AND\n" +
            "  (d.gender = ac.gender OR ac.gender IS NULL)\n" +
            ") AS " + CaseUtils.toCamelCase(AGE_CLASS_AT_TIME_COLUMN, true) + " \n" +
            "FROM \"${" + SCHEMA_VARIABLE + "}\".\"${" + QUERY_VARIABLE + "}\" c " +
            "LEFT JOIN \"${" + EHR_PATH_VARIABLE + "}\".study.demographics d ON (d.Id = c.${" + ID_COLUMN_VARIABLE + "})";


}
