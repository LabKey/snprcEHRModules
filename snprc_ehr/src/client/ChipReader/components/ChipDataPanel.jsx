/* eslint-disable no-await-in-loop */

import React from 'react'
import { requestPort, connect, close } from '../services/serialService'
import { getChipData } from '../services/microChipReader'
import constants from '../constants/index'

export default class ChipDataPanel extends React.Component {
    state = {
        isReading: false,
        connection: undefined
    }
componentDidMount = () => {
        console.log('ChipDataPanel mounted')
    }
onConnectClick = () => {
        if (!this.state.connection) {
            requestPort().then(serialPort => connect(serialPort, this.props.serialOptions).then(connection => {
                this.props.handleErrorMessage(undefined)
                this.props.handleSetConnection(connection)
            })).catch(error => {
                this.props.handleErrorMessage(error.message)
            })
        }
    }
flushPromises = () => { return new Promise(resolve => setTimeout(resolve)) }
onStartClick = async () => {
        if (this.state.connection && !this.state.isReading) {
            this.setState(prevState => (
                {
                    ...prevState,
                    isReading: true
                }
            ), async () => {
                this.props.handleDataChange(undefined)
                // let data

                console.log('starting reader')

                // serial port read loop
                while (this.state.isReading) {
                    const data = await getChipData(this.state.connection)
                        .then(results => {
                            // clear error message
                            if ((this.props.errorMessage && results && results.animalId)
                                || this.props.errorMessage === constants.timeOutErrorMessage) {
                                this.props.handleErrorMessage(undefined)
                            }
                            return results
                        })

                        .catch(error => {
                            this.props.handleErrorMessage(error.message)
                        })

                    await this.flushPromises() // interupt event loop

                    if (data && data.chipId.length > 0) {
                        this.props.handleDataChange(data)
                    }
                }
            })
        }
    }
onQuitClick = () => {
        // close serial connection
        if (this.state.connection) {
            this.setState(prevState => ( // shutdown reading before closing connection
                {
                    ...prevState,
                    isReading: false
                }
            ), () => {
                close(this.state.connection).then(() => {
                    this.props.handleSetConnection(undefined)
                }).catch(error => {
                    this.props.handleErrorMessage(error.message)
                })
            })
        }
    }
handleChange = () => { /* lint fix */ }
render() {
        this.state.connection = this.props.connection // && this.props.connection

        const { chipId, animalId, temperature } = this.props.chipData // && this.props.chipData

        return (
          <>
            <div className="wizard-panel__rows">
              <div className="wizard-panel__row">
                <div className="wizard-panel__col">
                  <div>
                    <label className="field-label">Chip ID</label>
                    <input
                      className="cage-input"
                      defaultValue={ chipId }
                      disabled
                      id="chip-chipId"
                      onChange={ this.handleChange }
                      placeholder="Chip ID"
                    />
                  </div>
                </div>
              </div>
              <div className="wizard-panel__row">
                <div className="wizard-panel__col">
                  <div>
                    <label className="field-label">Animal ID</label>
                    <input
                      className="cage-input"
                      defaultValue={ animalId && animalId.value }
                      disabled
                      id="chip-animalId"
                      onChange={ this.handleChange }
                      placeholder="Animal ID"
                    />
                  </div>
                </div>
              </div>
              <div className="wizard-panel__row">
                <div className="wizard-panel__col">
                  <div>
                    <label className="field-label">Temperature</label>
                    <input
                      className="cage-input"
                      defaultValue={ temperature }
                      disabled
                      id="chip-temperature"
                      onChange={ this.handleChange }
                      placeholder="Temperature"
                    />
                  </div>
                </div>
              </div>
            </div>

            <div className="summary-panel__rows">

              <div className="summary-panel__row">

                { // display reader state
                            (this.state.connection && !this.state.isReading && <div className="chip-info-span"> Reader connected - not reading  </div>)
                            || (this.state.isReading && <div className="chip-info-span">  Reader connected - reading </div>)
                            || <div className="chip-info-span"> Reader disconnected  </div>
                        }

                <div className="button-row">
                  <button id="connect" onClick={ this.onConnectClick } type="button">Connect to Reader</button>
                  <button id="start" onClick={ this.onStartClick } type="button">Start</button>
                  <button id="stop" onClick={ this.onQuitClick } type="button">Stop</button>
                </div>
              </div>
            </div>

          </>
        )
    }
}
