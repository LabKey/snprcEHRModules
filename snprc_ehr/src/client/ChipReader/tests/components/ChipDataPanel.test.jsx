/* eslint-disable no-unused-vars */

import React from 'react'
import { shallow } from 'enzyme'
import moment from 'moment'
import ChipDataPanel from '../../components/ChipDataPanel'
import constants from '../../constants'

jest.mock('../../services/serialService')

let wrapper

beforeEach(() => {
    wrapper = shallow(<ChipDataPanel
      chipData={ { chipId: undefined, animalId: undefined, temperature: undefined } }
      connection={ undefined }
      errorMessage={ undefined }
      handleDataChange={ () => { } }
      handleErrorMessage={ () => { } }
      handleSetConnection={ () => { } }
      serialOptions={ constants.serialOptions }
    />)
})

afterEach(() => {
    wrapper.unmount()
})

describe('ChipDataPanel tests', () => {
    test('Should render the ChipDataPanel.', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('Should connect to reader.', () => {
        const connectButton = wrapper.find('#connect')
        connectButton.simulate('click')
        wrapper.setProps({ connection: { port: {} } })
        expect(wrapper).toMatchSnapshot()
        const connectTextEl = wrapper.find('.chip-info-span')
        expect(connectTextEl.text().trim()).toEqual('Reader connected - not reading')
        connectTextEl.unmount()
    })

    test('Should start reading.', () => {
        wrapper.setProps({ connection: { port: {} } }) // test requires a connection object
        const startButton = wrapper.find('#start')
        startButton.simulate('click')
        expect(wrapper).toMatchSnapshot()
        const readTextEl = wrapper.find('.chip-info-span')
        expect(readTextEl.text().trim()).toEqual('Reader connected - reading')
        wrapper.setState({ isReading: false }) // cancel reading after tests (exit read loop)
    })

    test('Should stop reading.', () => {
      // start reading
      wrapper.setProps({ connection: { port: {} } }) // test requires a connection object
      wrapper.find('#start').simulate('click')
      expect(wrapper.find('.chip-info-span').text().trim()).toEqual('Reader connected - reading')

      // quit reading
      const stopButton = wrapper.find('#stop')
      stopButton.simulate('click')
      wrapper.setProps({ connection: undefined, isReading: false }) // close connection object

      expect(wrapper).toMatchSnapshot()
      expect(wrapper.find('.chip-info-span').text().trim()).toEqual('Reader disconnected')
  })
})
