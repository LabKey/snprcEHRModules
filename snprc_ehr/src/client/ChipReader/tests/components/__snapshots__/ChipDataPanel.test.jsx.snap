// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChipDataPanel tests Should connect to reader. 1`] = `
<Fragment>
  <div
    className="wizard-panel__rows"
  >
    <div
      className="wizard-panel__row"
    >
      <div
        className="wizard-panel__col"
      >
        <div>
          <label
            className="field-label"
          >
            Chip ID
          </label>
          <input
            className="cage-input"
            disabled={true}
            id="chip-chipId"
            onChange={[Function]}
            placeholder="Chip ID"
          />
        </div>
      </div>
    </div>
    <div
      className="wizard-panel__row"
    >
      <div
        className="wizard-panel__col"
      >
        <div>
          <label
            className="field-label"
          >
            Animal ID
          </label>
          <input
            className="cage-input"
            disabled={true}
            id="chip-animalId"
            onChange={[Function]}
            placeholder="Animal ID"
          />
        </div>
      </div>
    </div>
    <div
      className="wizard-panel__row"
    >
      <div
        className="wizard-panel__col"
      >
        <div>
          <label
            className="field-label"
          >
            Temperature
          </label>
          <input
            className="cage-input"
            disabled={true}
            id="chip-temperature"
            onChange={[Function]}
            placeholder="Temperature"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="summary-panel__rows"
  >
    <div
      className="summary-panel__row"
    >
      <div
        className="chip-info-span"
      >
         Reader connected - not reading  
      </div>
      <div
        className="button-row"
      >
        <button
          id="connect"
          onClick={[Function]}
          type="button"
        >
          Connect to Reader
        </button>
        <button
          id="start"
          onClick={[Function]}
          type="button"
        >
          Start
        </button>
        <button
          id="stop"
          onClick={[Function]}
          type="button"
        >
          Stop
        </button>
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`ChipDataPanel tests Should render the ChipDataPanel. 1`] = `
<Fragment>
  <div
    className="wizard-panel__rows"
  >
    <div
      className="wizard-panel__row"
    >
      <div
        className="wizard-panel__col"
      >
        <div>
          <label
            className="field-label"
          >
            Chip ID
          </label>
          <input
            className="cage-input"
            disabled={true}
            id="chip-chipId"
            onChange={[Function]}
            placeholder="Chip ID"
          />
        </div>
      </div>
    </div>
    <div
      className="wizard-panel__row"
    >
      <div
        className="wizard-panel__col"
      >
        <div>
          <label
            className="field-label"
          >
            Animal ID
          </label>
          <input
            className="cage-input"
            disabled={true}
            id="chip-animalId"
            onChange={[Function]}
            placeholder="Animal ID"
          />
        </div>
      </div>
    </div>
    <div
      className="wizard-panel__row"
    >
      <div
        className="wizard-panel__col"
      >
        <div>
          <label
            className="field-label"
          >
            Temperature
          </label>
          <input
            className="cage-input"
            disabled={true}
            id="chip-temperature"
            onChange={[Function]}
            placeholder="Temperature"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="summary-panel__rows"
  >
    <div
      className="summary-panel__row"
    >
      <div
        className="chip-info-span"
      >
         Reader disconnected  
      </div>
      <div
        className="button-row"
      >
        <button
          id="connect"
          onClick={[Function]}
          type="button"
        >
          Connect to Reader
        </button>
        <button
          id="start"
          onClick={[Function]}
          type="button"
        >
          Start
        </button>
        <button
          id="stop"
          onClick={[Function]}
          type="button"
        >
          Stop
        </button>
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`ChipDataPanel tests Should start reading. 1`] = `
<Fragment>
  <div
    className="wizard-panel__rows"
  >
    <div
      className="wizard-panel__row"
    >
      <div
        className="wizard-panel__col"
      >
        <div>
          <label
            className="field-label"
          >
            Chip ID
          </label>
          <input
            className="cage-input"
            disabled={true}
            id="chip-chipId"
            onChange={[Function]}
            placeholder="Chip ID"
          />
        </div>
      </div>
    </div>
    <div
      className="wizard-panel__row"
    >
      <div
        className="wizard-panel__col"
      >
        <div>
          <label
            className="field-label"
          >
            Animal ID
          </label>
          <input
            className="cage-input"
            disabled={true}
            id="chip-animalId"
            onChange={[Function]}
            placeholder="Animal ID"
          />
        </div>
      </div>
    </div>
    <div
      className="wizard-panel__row"
    >
      <div
        className="wizard-panel__col"
      >
        <div>
          <label
            className="field-label"
          >
            Temperature
          </label>
          <input
            className="cage-input"
            disabled={true}
            id="chip-temperature"
            onChange={[Function]}
            placeholder="Temperature"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="summary-panel__rows"
  >
    <div
      className="summary-panel__row"
    >
      <div
        className="chip-info-span"
      >
          Reader connected - reading 
      </div>
      <div
        className="button-row"
      >
        <button
          id="connect"
          onClick={[Function]}
          type="button"
        >
          Connect to Reader
        </button>
        <button
          id="start"
          onClick={[Function]}
          type="button"
        >
          Start
        </button>
        <button
          id="stop"
          onClick={[Function]}
          type="button"
        >
          Stop
        </button>
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`ChipDataPanel tests Should stop reading. 1`] = `
<Fragment>
  <div
    className="wizard-panel__rows"
  >
    <div
      className="wizard-panel__row"
    >
      <div
        className="wizard-panel__col"
      >
        <div>
          <label
            className="field-label"
          >
            Chip ID
          </label>
          <input
            className="cage-input"
            disabled={true}
            id="chip-chipId"
            onChange={[Function]}
            placeholder="Chip ID"
          />
        </div>
      </div>
    </div>
    <div
      className="wizard-panel__row"
    >
      <div
        className="wizard-panel__col"
      >
        <div>
          <label
            className="field-label"
          >
            Animal ID
          </label>
          <input
            className="cage-input"
            disabled={true}
            id="chip-animalId"
            onChange={[Function]}
            placeholder="Animal ID"
          />
        </div>
      </div>
    </div>
    <div
      className="wizard-panel__row"
    >
      <div
        className="wizard-panel__col"
      >
        <div>
          <label
            className="field-label"
          >
            Temperature
          </label>
          <input
            className="cage-input"
            disabled={true}
            id="chip-temperature"
            onChange={[Function]}
            placeholder="Temperature"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="summary-panel__rows"
  >
    <div
      className="summary-panel__row"
    >
      <div
        className="chip-info-span"
      >
         Reader disconnected  
      </div>
      <div
        className="button-row"
      >
        <button
          id="connect"
          onClick={[Function]}
          type="button"
        >
          Connect to Reader
        </button>
        <button
          id="start"
          onClick={[Function]}
          type="button"
        >
          Start
        </button>
        <button
          id="stop"
          onClick={[Function]}
          type="button"
        >
          Stop
        </button>
      </div>
    </div>
  </div>
</Fragment>
`;
